src/main/groovy/wooga/gradle/github/publish/internal/ReleaseAssetUpload.groovy

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Wooga GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package wooga.gradle.github.publish.internal
+
+import org.apache.tika.detect.Detector
+import org.apache.tika.metadata.Metadata
+import org.apache.tika.mime.MediaType
+import org.apache.tika.parser.AutoDetectParser
+import org.kohsuke.github.GHAsset
+import org.kohsuke.github.GHRelease
+import org.kohsuke.github.HttpException
+
+class ReleaseAssetUpload {
+    static GHAsset uploadAsset(GHRelease release, File assetFile) {
+        def contentType = getAssetContentType(assetFile)
+        FileInputStream stream = new FileInputStream(assetFile)
+        String fileName = URLEncoder.encode(assetFile.name, "UTF-8")
+
+        uploadAsset(release, fileName, stream, contentType)
+    }
+
+    static GHAsset uploadAsset(GHRelease release, String fileName, InputStream stream, String contentType) {
+        uploadAssetRetry(release, fileName, stream, contentType)
+    }
+
+    private static GHAsset uploadAssetRetry(GHRelease release, String fileName, InputStream stream, String contentType, int retryCount = 0) {
+        try {
+            release.uploadAsset(fileName, stream, contentType)
+        } catch(HttpException error) {
+            if (error.responseCode == 502 && retryCount < maxRetries) {
+                return uploadAssetRetry(release, fileName, stream, contentType, ++retryCount)
+            }
+
+            throw error
```

Comment:
Is it intended to throw for each failed upload attempt?

Reply:
Looking further down, I can answer this myself ;)

Reply:
see the `return` statement on line 46.
It calls itself recursively and returns the uploaded `GHAsset` item if one of the retries works.